cmd/aries-agent-mobile/pkg/wrappers/storage/storage.go

```diff
@@ -8,47 +8,366 @@ SPDX-License-Identifier: Apache-2.0
+				spi.ErrStoreNotFound)
+		}
+
+		return fmt.Errorf("failed to set store config in mobile binding provider: %w", err)
+	}
+
+	return nil
 }
 
-type store struct{ api.Store }
+// GetStoreConfig gets the current store configuration.
+// The store must be created prior to calling this method.
+// If the store cannot be found, then an error wrapping ErrStoreNotFound will be returned.
+// If name is blank, then an error will be returned.
+func (p *Provider) GetStoreConfig(name string) (spi.StoreConfiguration, error) {
+	configBytes, err := p.mobileBindingProvider.GetStoreConfig(name)
+	if err != nil {
+		// The errors returned from the mobile binding provider will just have the error text without any distinct
+		// error types like the ones defined in aries-framework-go/spi/storage. In order to still allow for the use of
+		// errors.Is(err, spi.ErrStoreNotFound) in higher level functions,
+		// we need to wrap spi.ErrStoreNotFound back in if we detect it.
+		if strings.Contains(err.Error(), spi.ErrStoreNotFound.Error()) {
+			return spi.StoreConfiguration{},
+				fmt.Errorf("failed to get store config from mobile binding provider: %s: %w", err.Error(),
+					spi.ErrStoreNotFound)
+		}
+
+		return spi.StoreConfiguration{},
+			fmt.Errorf("failed to get store config from mobile binding provider: %w", err)
+	}
+
+	var storeConfig spi.StoreConfiguration
+
+	err = json.Unmarshal(configBytes, &storeConfig)
+	if err != nil {
+		return spi.StoreConfiguration{}, fmt.Errorf("failed to unmarshal store configuration bytes: %w", err)
+	}
+
+	return storeConfig, nil
+}
+
+// GetOpenStores returns all currently open stores.
+func (p *Provider) GetOpenStores() []spi.Store {
+	p.lock.RLock()
+	defer p.lock.RUnlock()
+
+	openSPIStores := make([]spi.Store, len(p.openStores))
+
+	var counter int
+
+	for _, openStoreWrapper := range p.openStores {
+		openSPIStores[counter] = openStoreWrapper
+		counter++
+	}
+
+	return openSPIStores
+}
+
+// Close closes all stores created under this store provider.
+// For persistent mobile binding store implementations, this does not delete any data in the underlying databases.
+func (p *Provider) Close() error {
+	p.lock.RLock()
+
+	openStoresSnapshot := make([]*store, len(p.openStores))
+
+	var counter int
+
+	for _, openStore := range p.openStores {
+		openStoresSnapshot[counter] = openStore
+		counter++
+	}
+	p.lock.RUnlock()
+
+	for _, openStore := range openStoresSnapshot {
+		err := openStore.Close()
+		if err != nil {
+			return fmt.Errorf("failed to close store: %w", err)
+		}
+	}
+
+	return nil
+}
+
+func (p *Provider) removeStore(name string) {
+	p.lock.Lock()
+	defer p.lock.Unlock()
+
+	delete(p.openStores, name)
+}
+
+type store struct {
+	mobileBindingStore api.Store
+	name               string
+	close              closer
+}
+
+func (s *store) Put(key string, value []byte, tags ...spi.Tag) error {
+	var tagsBytes []byte
+
+	if len(tags) > 0 {
+		var err error
```

Comment:
`var err error` can be moved out of this block

Reply:
Indeed it could...is there an advantage to doing so?

Reply:
Done in #2567!